src/js/port_handler.js

```diff
@@ -174,9 +174,15 @@ PortHandler.detectPort = function(currentPorts) {
         if (GUI.auto_connect && !GUI.connecting_to && !GUI.connected_to) {
             // start connect procedure. We need firmware flasher protection over here
             if (GUI.active_tab !== 'firmware_flasher') {
+                let connectionTimeout = 100;
+                ConfigStorage.get('connectionTimeout', function (result) {
+                    if (result.connectionTimeout) {
+                        connectionTimeout = result.connectionTimeout;
+                    }
+                });
                 GUI.timeout_add('auto-connect_timeout', function () {
```

Comment:
Maybe we need to move the `GUI.timeout_add` inside the `ConfigStorage.get()` callback to be sure that the `connectionTimeout` has been read?